src/structure_managers/cluster_ref_key.hh

```diff
@@ -41,140 +41,125 @@
+   * `index_sequence`.
+   *
+   * @tparam Index
+   * @param index_sequence
+   *
+   * @return returns the `Index` elment in the index sequence.
+   */
+  template <size_t Index, size_t... Ints>
+  constexpr size_t
+  get_index(const std::index_sequence<Ints...> & /*index_sequence*/) {
+    return std::get<Index>(std::make_tuple(Ints...));
+  }
 
-  /* ---------------------------------------------------------------------- */
-  //! returns the cluster layer for accessing properties at a specific layer in
-  //! a stack
+  /**
+   * From the index sequence given by the type of `layers_by_order` the layer is
+   * returned at order/position `Order`.
+   *
+   * @tparam Order
+   * @param layers_by_order the layers of each order in an index sequence
+   *
+   * @return returns the layer at the order `Order`
+   */
   template <size_t Order, size_t... Ints>
-  constexpr size_t compute_cluster_layer(const std::index_sequence<Ints...> &) {
-    using ActiveDimensions =
-        typename internal::HeadExtractor<Order, std::index_sequence<>,
-                                         Ints...>::type;
-    return internal::MinExtractor<ActiveDimensions>::value;
+  constexpr size_t
+  get_layer(const std::index_sequence<Ints...> & layers_by_order) {
+    static_assert(Order > 0, "Order is <1 this should not be.");
+    static_assert(Order <= sizeof...(Ints),
+                  "Order should be within the MaxOrder.");
+    return get_index<Order - 1>(layers_by_order);
   }
 
-  // #BUG8486@(all) removed the MaxOrder template parameter, and the meaning
-  // of the access index was not clear, changed name to order
+  /**
+   * Returns the index sequence given by the `Ints` as array.
+   *
+   * @param layers_by_order the type captures the layers of each order in an
+   *        index sequence.
+   *
+   * @return returns `Ints` as array
+   */
   template <size_t... Ints>
-  constexpr size_t get_layer(const size_t order,
-                             const std::index_sequence<Ints...>) {
-    constexpr size_t arr[]{Ints...};
-    return arr[order - 1];
+  constexpr std::array<size_t, sizeof...(Ints)>
+  get_layers(const std::index_sequence<Ints...> & /*layers_by_order*/) {
+    return std::array<size_t, sizeof...(Ints)>{Ints...};
   }
 
   /**
-   * Static access to layer by cluster dimension (e.g., for defining template
-   * parameter `NbRow` for a property
+   * Returns the smallest number of layers up to the order `ActiveMaxOrder` in
+   * the `LayersByOrder` index sequence. It is needed to access properties at a
+   * specific layer in a manager stack.
+   *
+   * @tparam ActiveMaxOrder the maximum order to check the `LayersByOrders` for
+   *         the minimum value.
+   * @param layers_by_order the type captures the layers of each order in an
+   *        index sequence.
+   *
+   * @return the smallest value in the given index sequence up to Order elements
    */
-  template <size_t index, size_t... Ints>
```

Comment:
renamed the `get` function to `get_index` because the compiler confused this sometime with std::get